src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/OutputJsonWriter.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.config.OutputFieldType;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+
+public class OutputJsonWriter implements OutputWriter {
+
+    private final Map<String, OutputFieldJsonWriter> writers;
+    private final Serializer<JsonNode> serializer;
+    private ArrayNode root;
+
+    OutputJsonWriter(final Map<String, OutputFieldJsonWriter> writers) {
+
+        this.writers = writers;
+        serializer = new JsonLinesSerializer();
+    }
+
+    public void writeRecord(final SinkRecord record,
+                            final OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(record, "record cannot be null");
+        Objects.requireNonNull(outputStream, "outputStream cannot be null");
+        if (root == null) {
+            root = JsonNodeFactory.instance.arrayNode();
+        }
+        root.add(writeFields(record));
+    }
+
+    public void writeLastRecord(final SinkRecord record,
+                                final OutputStream outputStream) throws IOException {
+        Objects.requireNonNull(record, "record cannot be null");
+        Objects.requireNonNull(outputStream, "outputStream cannot be null");
+
+        writeRecord(record, outputStream);
+        try {
+            outputStream.write(serializer.serialize(record.topic(), root));
+        } catch (final SerializationException e) {
+            throw new DataException("Converting Kafka Connect data to byte[] failed due to serialization error: ", e);
+        }
+        root = null;
+    }
+
+    private JsonNode writeFields(final SinkRecord record) throws IOException {
+        final Iterator<Map.Entry<String, OutputFieldJsonWriter>> writerIter = writers.entrySet().iterator();
+
+        if (writers.size() == 1) {
```

Comment:
Here's some inconsistency: 2-or-more-field writers produce an output with a wrapper object, 1-field writers prodice an output without a wrapper object. Wrapper or not should be configurable and independent from the number of output fields.

Reply:
Tests are already there, so, is it Ok, if I add Simple Output as a private variable?

Reply:
Nevermind. It is easier to extend it later. Simple configuration doesn't look right with a current Builder and required well-thought-out configuration (e.g What happened if user specified only offset as a output value. It is possible, but does it make sense?)

Reply:
- [ ] Remove 1-field writer